src/irr/asset/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -383,17 +411,29 @@ void main()
     case 6:
     case 7:
     case 9:
-        d = Fresnel_dielectric(PC.params.Ni, bsdfParams.NdotL);
+    {
+        //assuming light_pos=eye_pos, V=H so VdotH=1.0
+        //in order for this to work fine (without this assumption), user must override main() -- bad
+        //another solution would be to somehow return alpha from irr_computeLighting()
+        float VdotH = 1.0;
+        d = Fresnel_dielectric(PC.params.Ni, VdotH);
+    }
         break;
     default:
         if ((PC.params.extra&(map_d_MASK)) == (map_d_MASK))
             d = texture(map_d, UV).r;
```

Comment:
if you use a non-fresnel alpha (comes from `params.d` or texture) then I'd modulate the ambient in the light eval and the diffuse in the `brdf_cos_eval` by it (even though that approach is not exactly PBR) in all the lighting models.

Reply:
just multiply `color` by it here.